### PR TITLE
Doc for df getitem

### PR DESCRIPTION
**Pull request by Rubtsowa**

### CODE REVIEW

File: sdc/datatypes/hpat_pandas_dataframe_functions.py [hunk 1]
```diff
@@ -1600,6 +1600,98 @@ def _df_getitem_bool_array_idx_impl(self, idx):
...
+        Pandas DataFrame method :meth:`pandas.DataFrame.getitem` implementation.
+        .. only:: developer
+
+        Test: python -m sdc.runtests -k sdc.tests.test_dataframe.TestDataFrame.test_df_getitem*
+
+        Parameters
+        -----------
+        self: :obj:`pandas.DataFrame`
+            input arg
+        idx: :obj:`pandas.Series`, :obj:`string`, :obj:`slice`, :obj:`tuple`, :obj:`array`, :obj:`tuple`
```
Comment (densmirn) [inline sdc/datatypes/hpat_pandas_dataframe_functions.py]:
```suggestion
        idx: :obj:`pandas.Series`, :obj:`string`, :obj:`slice`, :obj:`tuple`, :obj:`array`
```

File: sdc/datatypes/hpat_pandas_dataframe_functions.py [hunk 2]
```diff
@@ -1600,6 +1600,98 @@ def _df_getitem_bool_array_idx_impl(self, idx):
...
+            input arg
+
+        Returns
+        -------
+        :obj: `pandas.DataFrame`
```
Comment (densmirn) [inline sdc/datatypes/hpat_pandas_dataframe_functions.py]:
```suggestion
        :obj:`pandas.DataFrame`
```

File: sdc/datatypes/hpat_pandas_dataframe_functions.py [hunk 3]
```diff
@@ -1600,6 +1600,98 @@ def _df_getitem_bool_array_idx_impl(self, idx):
...
+
+        Limitations
+        -----------
+        Supported ``key`` can be one of the following:
+            - String literal, e.g. :obj:`df['A']`
+            - A slice, e.g. :obj:`df[2:5]`
+            - A tuple of string, e.g. :obj:`df[('A', 'B')]`
+            - An array of booleans, e.g. :obj:`df[True,False]`
+            - A series of booleans, e.g. :obj:`df(series([True,False]))`
+        Supported getting a column through getting attribute.
```
Comment (densmirn) [inline sdc/datatypes/hpat_pandas_dataframe_functions.py]:
```suggestion
        Supported ``key`` can be one of the following:

        * String literal, e.g. :obj:`df['A']`
        * A slice, e.g. :obj:`df[2:5]`
        * A tuple of string, e.g. :obj:`df[('A', 'B')]`
        * An array of booleans, e.g. :obj:`df[True,False]`
        * A series of booleans, e.g. :obj:`df(series([True,False]))`

        Supported getting a column through getting attribute.
```

File: sdc/datatypes/hpat_pandas_dataframe_functions.py [hunk 4]
```diff
@@ -1600,6 +1600,114 @@ def _df_getitem_bool_array_idx_impl(self, idx):
 
 @sdc_overload(operator.getitem)
 def sdc_pandas_dataframe_getitem(self, idx):
+    """
+        Intel Scalable Dataframe Compiler User Guide
+        ********************************************
+        Pandas API: pandas.DataFrame.getitem
+
+        Get column(s) of DataFrame by key.
```
Comment (kozlov-alexey) [inline sdc/datatypes/hpat_pandas_dataframe_functions.py]: This is not accurate - examples show that df[1:3] selects rows from the df, not columns. Maybe:
```suggestion
        Get data from a DataFrame by indexer.
```

**Pull request merged**